servicetalk-concurrent-api/docs/modules/ROOT/pages/blocking-implementation.adoc

```diff
@@ -38,22 +40,60 @@ Part 2. defines all the interactions using the ReactiveStreams specifications, i
-its operators, and the `Executor`.
-
-image::executor-thread-selection.png[Executor inference]
-
-The above interaction provides all operators and the source an opportunity to piggy-back on the same `Executor` for all
-interactions with the `Subscriber`. Any operator in a chain has two options while accepting a `Subscriber`:
-
-- Wrap the `Subscriber` such that all `Subscriber` and `Subscription` method calls are offloaded to a chosen
-`Executor`. This mode is for operators that process signals from the original `Publisher` asynchronously.
-- Do not wrap the `Subscriber` and forward all `Subscriber` and `Subscription` methods directly on the calling thread to
-avoid the overhead of wrapping each and every `Subscriber` and `Subscription` in a chain. This mode is for operators
-that process signals from the original `Publisher` synchronously.
+== Task Offloading
+
+ServiceTalk uses Netty for network I/O and the Netty implementation uses a fixed number of I/O threads for executing
+network operations. To maximize throughput and minimize latency it is important to ensure that at least one I/O thread
+is always ready to respond immediately to network activity. One approach to ensure I/O thread availability is to
+carefully limit the scope of work done by I/O threads and, when practical, perform any other necessary work on some
+other non-precious thread. Moving work from I/O threads to other threads is called “offloading” and is a core technique
+used by ServiceTalk.
+
+ServiceTalk will, by default, execute most application code on an offloaded “safe” thread. For efficiency, some
+interfaces are invoked synchronously and must not block. Usually these interfaces are functional interfaces that are
+meant to be low–overhead pure functions. These interfaces are explictly about the non-blocking requirement. Examples
+of application code which must not block and may execute on event loop thread include the `Predicate` provided to the `appendServiceFilter(Predicate<StreamingHttpRequest>,
+StreamingHttpServiceFilterFactory)` method of link:{source-root}/servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java[HttpServerBuilder].
+
+For most invocations of application code, if the application developer knows that their code cannot block and always
+executes quickly in near constant time they can request that ServiceTalk not offload their code. This will improve
+application performance by reducing latency and overhead. Requests to not offload will be honored by ServiceTalk if all
+the other components in the same execution path have also opted out of offloading. As a last resort, work tasks may also
+be queued to be performed as threads are available.
+
+ServiceTalk is designed to be fully asynchronous except where the API provides explicit blocking behaviour as a
+convenience. Offloading is the execution of program logic code using a thread (or equivalent) other than the application
```

Comment:
`Offloading is the execution...` - it seems to me that this explanation should appear much earlier. I suppose after the first paragraph of this section. Consider reordering paragraphs to emphasize the high-level concepts first and then dive into details, such as synchronous execution of functional interfaces (current 2nd paragraph).

Reply:
I moved the `Offloading is the execution…` paragraphs to `performance.adoc#ExecutionStrategy` section which seems like a better location for the material.